internals/secrethub/tpl/v2_test.go

```diff
@@ -1,23 +1,469 @@
+		"secret path in middle": {
+			input: "hello {{path/to/secret}} secret",
+			expected: []node{
+				character('h'),
+				character('e'),
+				character('l'),
+				character('l'),
+				character('o'),
+				character(' '),
+				secret{
+					path: []node{
+						character('p'),
+						character('a'),
+						character('t'),
+						character('h'),
+						character('/'),
+						character('t'),
+						character('o'),
+						character('/'),
+						character('s'),
+						character('e'),
+						character('c'),
+						character('r'),
+						character('e'),
+						character('t'),
+					},
+				},
+				character(' '),
+				character('s'),
+				character('e'),
+				character('c'),
+				character('r'),
+				character('e'),
+				character('t'),
+			},
+		},
+		"variable in secret path at start": {
+			input: "{{${var}/secret}}",
+			expected: []node{
+				secret{
+					path: []node{
+						variable{
+							key: "var",
+						},
+						character('/'),
+						character('s'),
+						character('e'),
+						character('c'),
+						character('r'),
+						character('e'),
+						character('t'),
+					},
+				},
+			},
+		},
+		"variable in secret path at end": {
+			input: "{{secret${var}}}",
+			expected: []node{
+				secret{
+					path: []node{
+						character('s'),
+						character('e'),
+						character('c'),
+						character('r'),
+						character('e'),
+						character('t'),
+						variable{
+							key: "var",
+						},
+					},
+				},
+			},
+		},
+		"variable in secret path at end with space": {
+			input: "{{ secret${var} }}",
+			expected: []node{
+				secret{
+					path: []node{
+						character('s'),
+						character('e'),
+						character('c'),
+						character('r'),
+						character('e'),
+						character('t'),
+						variable{
+							key: "var",
+						},
+					},
+				},
+			},
+		},
+		"variable in secret path in middle": {
+			input: "{{path/to/${var}/secret}}",
+			expected: []node{
+				secret{
+					path: []node{
+						character('p'),
+						character('a'),
+						character('t'),
+						character('h'),
+						character('/'),
+						character('t'),
+						character('o'),
+						character('/'),
+						variable{
+							key: "var",
+						},
+						character('/'),
+						character('s'),
+						character('e'),
+						character('c'),
+						character('r'),
+						character('e'),
+						character('t'),
+					},
+				},
+			},
+		},
+		"variable with spaces": {
+			input: "${ var }",
+			expected: []node{
+				variable{
+					key: "var",
+				},
+			},
+		},
+		"secret with spaces": {
+			input: "{{ path/to/secret }}",
+			expected: []node{
+				secret{
+					path: []node{
+						character('p'),
+						character('a'),
+						character('t'),
+						character('h'),
+						character('/'),
+						character('t'),
+						character('o'),
+						character('/'),
+						character('s'),
+						character('e'),
+						character('c'),
+						character('r'),
+						character('e'),
+						character('t'),
+					},
+				},
+			},
+		},
+		"{ and } chars used": {
```

Comment:
How about adding a case with `{` and `}` characters wrapping a secret tag, e.g. `{{{path/to/secret}}}`?

Reply:
3196cc06cae09261e54eb0d39d23c21d1ed95f94